GUACAMOLE-1692: Updated german translation

sirux88 opened the pull request:
some improvements to german translation

IT-Experte commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -651,16 +769,23 @@
+        "FIELD_HEADER_SFTP_PORT"                  : "Port:",
+        "FIELD_HEADER_SFTP_PRIVATE_KEY"           : "Privater Schlüssel:",
+        "FIELD_HEADER_SFTP_ROOT_DIRECTORY"        : "SFTP-Wurzelverzeichnis:",
+        "FIELD_HEADER_SFTP_DISABLE_UPLOAD"        : "Disable file upload:",
```
Translation "Hochladen Deaktivieren"

sirux88 replied:
added the proposal

IT-Experte commented:
Unfortunately I haven't had time for the translation in the last few months. Very good work.

IT-Experte reviewed (changes requested):
Maybe some change should be made.

sirux88 commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -147,6 +150,20 @@
+    "COLOR_SCHEME" : {
+
+        "ACTION_HIDE_DETAILS" : "Anzeigen",
+        "ACTION_SHOW_DETAILS" : "Ausblenden",
```
Yeah you're right. I swap both

sirux88 commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -470,7 +565,7 @@
         "SECTION_HEADER_DISPLAY"            : "Bildschirm",
         "SECTION_HEADER_GATEWAY"            : "Remote Desktop Gateway",
-        "SECTION_HEADER_LOAD_BALANCING"     : "Load Balancing",
+        "SECTION_HEADER_LOAD_BALANCING"     : "Lastausgleich",
```
Yeah, seems better to me aswell

sirux88 commented:
@IT-Experte: 
I oriented myself for all these `FIELD_HEADER_DISABLE_COPY` strings at the existing one within `RDP` block.
But I wasn't satisfied with the translation with `RDP` aswell but wanted to maintain the same kind of "speech".

I liked you're proposal so I inserted it everywhere you suggested and aswell within the `RDP` block

sirux88 commented:
I rebased to current master. So maybe it will get merged soon.

sirux88 commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -117,6 +120,10 @@
 
+        "MESSAGE_DEFAULT"      : "",
+        "MESSAGE_USER_JOINED"  : "Benutzer {ARGS_1} ist der Verbindung beigetreten.",
+        "MESSAGE_USER_LEFT"    : "Benutzer {ARGS_1} hat die Verbindung verlassen",
```
@IT-Experte:
I translated these additional strings since my initial commit

IT-Experte commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -361,7 +382,60 @@
+
+    "PLAYER" : {
+
+        "INFO_LOADING_RECORDING" : "Die Aufnahme wird momentan geladen. Bitte warten...",
```
A suggestion for shorter text: "Aufnahme wird geladen. Bitte warten..."

IT-Experte commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -361,7 +382,60 @@
+        "FIELD_HEADER_CONTAINER"       : "Container Name:",
+        "FIELD_HEADER_CREATE_RECORDING_PATH"  : "Aufname-Pfad automatisch erstellen:",
+        "FIELD_HEADER_CREATE_TYPESCRIPT_PATH" : "Typescript-Pfad automatisch erstellen:",
+        "FIELD_HEADER_EXEC_COMMAND"    : "Befehl (exec):",
```
A suggestion: "Befehl ausführen (exec)"

IT-Experte commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -399,6 +480,7 @@
         "FIELD_HEADER_IGNORE_CERT"     : "Serverzertifikat ignorieren:",
         "FIELD_HEADER_INITIAL_PROGRAM" : "Startprogramme:",
         "FIELD_HEADER_LOAD_BALANCE_INFO" : "Lastverteilungs-Info/Cookie:",
+        "FIELD_HEADER_NORMALIZE_CLIPBOARD" : "Zeilenende:",
```
It has to do with the character on the clipboard? "Zwischenablage normalisieren / Zeilenumbruch"

IT-Experte reviewed (approved):
Everything looks good. I have one more suggestion, but it is not important.

necouchman commented:
@sirux88 and @IT-Experte Is this translation ready? Or do any of @IT-Experte changes need to be addressed, first? I'm sorry, I don't know German, so I cannot make any determination between the suggestions and the original.

sirux88 commented:
I rebased to current master and updated the translation for the main application and TOTP extension to fit the latest language string @necouchman 

It should be fine to merge.

necouchman commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -21,12 +22,14 @@
         "ACTION_UPDATE_PASSWORD"    : "Passwort aktualisieren",
         "ACTION_VIEW_HISTORY"       : "Verlauf",
-
+        "ACTION_VIEW_RECORDING"     : "Ansicht",
```
Can you add a space below this line?

sirux88 replied:
done

necouchman commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -446,21 +538,28 @@
         "FIELD_OPTION_SECURITY_RDP"   : "RDP Verschlüsselung",
         "FIELD_OPTION_SECURITY_TLS"   : "TLS Verschlüsselung",
 
-        "FIELD_OPTION_SERVER_LAYOUT_DA_DK_QWERTY": "Dänisch (Qwerty)",
```
Any particular reason this was re-organized and moved down below? I think it should still be at the top?

sirux88 replied:
It should be on top yes.
But since it is down on `en.json` I aligned it to keep comparing easier.

If you want I move it up on both `en.json` and `de.json`

necouchman replied:
Sure, that would be fine if you want to fix it in both locations.

sirux88 replied:
I fixed it in all langugaes where it was present.

necouchman commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -580,9 +693,15 @@
         "FIELD_HEADER_RECORDING_INCLUDE_KEYS"   : "Tastendrücke aufzeichnen:",
         "FIELD_HEADER_RECORDING_NAME" : "Aufzeichnungsname:",
         "FIELD_HEADER_RECORDING_PATH" : "Aufzeichnungspfad:",
+
```
I think this extra space can be removed - the rest of the `FIELD_HEADER` items don't seem to be broken up?

sirux88 replied:
I removed it. Rest seems fine yeah

necouchman commented on guacamole/src/main/frontend/src/translations/de.json:
```diff
@@ -759,19 +889,21 @@
         "HELP_INPUT_METHOD_OSK"     : "@:CLIENT.HELP_INPUT_METHOD_OSK",
         "HELP_INPUT_METHOD_TEXT"    : "@:CLIENT.HELP_INPUT_METHOD_TEXT",
-        "HELP_LANGUAGE"             : "Um die Spracheinstellungen von Guacamole zu ändern,  wählen Sie eine der verfügbaren Sprachen.",
+        "HELP_LOCALE"             : "Um die Spracheinstellungen von Guacamole zu ändern,  wählen Sie eine der verfügbaren Sprachen.",
```
Maybe add one space to line up this `:` with the other lines above/below?

sirux88 replied:
Done

necouchman reviewed (changes requested):
Overall looks very good - just a few minor nitpicks on style and I think it'll be good to merge.

sirux88 commented:
I reworked according to your comments @necouchman. 
Additionally I added some more translation strings

sirux88 commented:
Let me know if I shall smash it into a single commit @necouchman

necouchman reviewed (approved):
Looks good - I also did a manual build and load of the code just to make sure everything was good, since the Github automated builds are failing.

sirux88 commented:
I made a copy/paste error here @necouchman 

https://github.com/apache/guacamole-client/blob/b8cc74c90e330bb97d910ece74b884fa3443e50c/guacamole/src/main/frontend/src/translations/en.json#L671
This should be `Danish` not `Dänisch` (what is Danish in German)

How shall we fix this? Any chance to do it without opening an JIRA issue?

necouchman commented:
@sirux88 No need for a new Jira issue, just open a new PR with the same Jira tags (GUACAMOLE-1692). I'll re-open that Jira issue.
